src/snowflake/connector/cache.py

```diff
@@ -494,7 +514,8 @@ def __getitem__(self, k: K) -> V:
 
     def _setitem(self, k: K, v: V) -> None:
         super()._setitem(k, v)
-        self._save_if_should()
+        if self._try_saving_when_set_item:
```

Comment:
I think this check could be moved into `_save_if_should`

Reply:
`_save_if_should` is also used by `clear_expired_entries`, so I do want `_try_saving_when_set_item` to impact the behavior of `clear_expired_entries`. or we could introduce another parameter in `_save_if_should` to control the behavior